src/components/TodoItem.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { FaTrash } from 'react-icons/fa';
+import { AiFillEdit } from 'react-icons/ai';
+import { useTodosContext } from '../context/TodosContext';
+import { useAuthContext } from '../context/AuthContext';
+import styles from '@/styles/TodoItem.module.css';
+
+const TodoItem = ({ itemProp }) => {
+  const [editing, setEditing] = useState(false);
+
+  const { handleChange, delTodo, setUpdate } = useTodosContext();
+  const { user } = useAuthContext();
+
+  const editInputRef = useRef(null);
+
+  const completedStyle = {
+    fontStyle: 'italic',
+    color: '#595959',
+    opacity: 0.4,
+    textDecoration: 'line-through',
+  };
+
+  const handleEditing = () => {
+    setEditing(true);
+  };
+
+  const handleUpdatedDone = (event) => {
+    if (event.key === 'Enter') {
+      setUpdate(editInputRef.current.value, itemProp.id);
+      setEditing(false);
+    }
+  };
+
+  return (
+    <li className={styles.item}>
+      <div className={styles.content}>
+        <input
+          type="checkbox"
+          checked={itemProp.completed}
+          onChange={() => handleChange(itemProp.id)}
+        />
+        {user && (
+          <button type="button" onClick={handleEditing}>
+            <AiFillEdit style={{ color: '#5e5e5e', fontSize: '16px' }} />
+          </button>
+        )}
+        <button type="button" onClick={() => delTodo(itemProp.id)}>
+          <FaTrash style={{ color: '#5e5e5e', fontSize: '16px' }} />
+        </button>
```

Comment:
The editing and deleting button now is before the text, to match the To-Do design and make your app looks better, kindly move them to the end of the parent div so they appear after the text.

![1](https://github.com/ClaudiaRojasSoto/React-ToDo-App/assets/107482707/79e77517-19f0-4aa8-bcf4-f56b26692a1e)

To get a better design result you can also move them to the right side of their container by using CSS (position relative - position  absolute - right) and by giving text left padding.

![2](https://github.com/ClaudiaRojasSoto/React-ToDo-App/assets/107482707/da635680-58d5-4a04-84d4-b2174a596414)

Reply:
Thank you very much @Abdelaziz-Ali Abdelaziz-Ali, I think you are very right. Moving the edit and delete icon improves the user experience. However, despite the fact that you gain in design by putting the icons at the end, at the same height as the input, I think that this does not improve the user experience because you must move the mouse to the right to make the changes, on the other hand, if They are there, so to speak, at hand, it is easier to delete or edit. I am waiting for your comments because the changes in the first part have already been made.

@ClaudiaRojasSoto